ank/doc/swdesign/README.md

```diff
@@ -53,6 +53,37 @@ Considered alternatives:
 * keep on using the environment logger
 * use another crate which provides tracing functions
 
+#### The CLI does not limit the size of the missed message buffer
+
+While the CLI waits for a response from the Ankaios server,
+it stores all other messages received from the Ankaios server in a buffer,
+as some messages might be needed after the response is received.
+If the Ankaios server takes time to response to the request and many other messages are send,
+the buffer can grow without limit.
+
+This problem could be solved by:
+
+* Introducing a ring buffer.
+  If the buffer grows to large, the oldest messages would be dropped.
+  This could lead to important messages being missed.
+* The size of the buffer could be limited.
+  If the buffer would exceed the limit, the CLI would exit with a failure.
+* The CLI could timeout, if the Ankaios server takes to long to respond.
+  This would not solve the problem directly, but it would reduce the probability of the problem occurring.
+
+None of the solution is implemented as:
+
+* All solution need an additional parameter (size of buffer or timeout)
+* This problem is unlikely to occur:
+  (1) the server take a long time to respond or does not respond at all.
+  (2) the server send many messages to the CLI
+* The CLI is intended for interactive usage.
+  If the response takes to long, the user can terminate the CLI.
+* The CLI is intended for use during development and not for use in production.
+  Crashes are more acceptable here.
+* The CLI will be used on a developer PC and not on embedded devices.
+  Memory constraints are not of the biggest concern.
+
```

Comment:
I rearranged a bit to fit it to the design decision structure normally used in this and the other documents:

```suggestion
#### The CLI does not limit the size of the missed message buffer
`swdd~cli-does-not-limit-missed-message-buffer-size~1`

Status: approved

While the CLI waits for a response from the Ankaios server,
it stores all other messages received from the Ankaios server in a buffer,
as some messages might be needed after the response is received.
If the Ankaios server takes time to response to the request and many other messages are send,
the buffer can grow without limit.

Rationale:

The buffer is left to grow without limit for the following reasons:

* This problem is unlikely to occur as:
  (1) the server must take a long time to respond or does not respond at all.
  (2) the server must send many messages to the CLI
* The CLI is intended for interactive usage.
  If the response takes to long, the user can terminate the CLI.
* The CLI is intended for use during development and not for use in production.
  Crashes are more acceptable here.
* The CLI will be used on a developer PC and not on embedded devices.
  Memory constraints are not of the biggest concern.
* All alternatives need an additional parameter (size of buffer or timeout)

Considered alternatives:

* Introducing a ring buffer.
  If the buffer grows to large, the oldest messages would be dropped.
  This could lead to important messages being missed.
* The size of the buffer could be limited.
  If the buffer would exceed the limit, the CLI would exit with a failure.
* The CLI could timeout, if the Ankaios server takes to long to respond.
  This would not solve the problem directly, but it would reduce the probability of the problem occurring.

```